controllers/openshift_controller.go

```diff
@@ -782,7 +782,7 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigInstallRequest() (ctrl.
 		// Update node selector in machine config pool with value from kataconfig instance
 		r.Log.Info("Updating machine config pool")
 		if foundMcp != nil {
-			if foundMcp.Spec.NodeSelector != nil {
+			if foundMcp.Spec.NodeSelector != nil && r.kataConfig.Spec.KataConfigPoolSelector != nil {
```

Comment:
@jensfr Could you remind me why we test for `foundMcp.Spec.NodeSelector` being non-nil? Looking at it now, it seems to me that if `r.kataConfig.Spec.KataConfigPoolSelector` is non-nil we should replace `foundMcp.Spec.NodeSelector`'s value no matter the previous value. (?) Also, what if a user had a KataConfigPoolSelector previously but now they want kata on all of their workers so they just delete the selector - is that case handled?

Reply:
@pmores I think you're right,the only check neccessary is if KataConfigPoolSelector is not nil. 

The other case is covered because we always create a kata-oc mcp and the default label used is `nodeSelector := &metav1.LabelSelector{MatchLabels: map[string]string{"node-role.kubernetes.io/worker": ""}}`. So when the nodeselector is removed from kataconfig.spec we will use this as default and the kata-oc mcp now contains all workers.